Task_1_1_2/src/main/java/ru/nsu/fit/rafik/task_1_1_2/MyStack.java

```diff
@@ -0,0 +1,98 @@
+package ru.nsu.fit.rafik.task_1_1_2;
+
+public class MyStack<A> {
+
+    // number of the element in the stack
+    private int count = 0;
+
+
+    // stack elements themselves
+    private Object[] values;
+
+    // size
+    private int size = 10000;
+
+    public MyStack() {
+        this.values = new Object[size];
+    }
+
+    // the stack is empty or not ?
+    public boolean isEmpty() {
+        if (count == 0) {
+            return true;
+        } else
+            return false;
+    }
+
+
+    // if there is any free space on the stack and adds a new element
+    public void push(A element) {
+        if (count < size) {
+            values[count] = element;
+            count++;
+        } else {
+            this.increaseStackSize();
+            throw new ArrayIndexOutOfBoundsException("stack overflow");
+        }
+    }
+
+    private void increaseStackSize() {
+        int newSize = this.size * 2;
+        Object[] newValues = (A[]) new Object[newSize];
+
+        for (int i = 0; i < this.size; i++) {
+            newValues[i] = this.values[i];
+        }
+
+            this.values = newValues;
+            this.size = newSize;
+        }
+    public void pushStack(MyStack anotherStack) {
+
+        int anotherStackCount = anotherStack.count;
+        A[] arr = (A[]) new Object[anotherStackCount];
+
+        for (int i = (anotherStackCount - 1); i >= 0; i--) {
+            arr[i] = (A) anotherStack.pop();
```

Comment:
You have access to inner stack structure, use arrays manipulation such as arraycopy